src/timer/delay.rs

```diff
@@ -114,22 +114,28 @@ pub type DelayMs<TIM> = FDelay<TIM, 1_000>;
 
 impl<TIM: Instance, const FREQ: u32> FDelay<TIM, FREQ> {
     /// Sleep for given time
-    pub fn delay(&mut self, time: TimerDurationU32<FREQ>) -> Result<(), Error> {
```

Comment:
I'm not too excited about this undocumented breaking API change, maybe it would be better to keep the `Result` for now?

Reply:
This version of Delay was added in #423 and not released yet.
So if now it is infallible what sense to keep Result ?

Reply:
Ah, I missed that. If all of the fallible API was just recently added, that's of course fine to change it back to infallible.